core/src/banking_stage.rs

```diff
@@ -1982,18 +1979,27 @@ impl BankingStage {
     fn receive_until(
         verified_receiver: &CrossbeamReceiver<Vec<PacketBatch>>,
         recv_timeout: Duration,
-        batching_timeout: Duration,
-        batch_size_upperbound: usize,
+        packet_count_upperbound: usize,
     ) -> Result<Vec<PacketBatch>, RecvTimeoutError> {
         let start = Instant::now();
         let mut packet_batches = verified_receiver.recv_timeout(recv_timeout)?;
+        let mut num_packets_received: usize =
+            packet_batches.iter().map(|batch| batch.packets.len()).sum();
         while let Ok(packet_batch) = verified_receiver.try_recv() {
-            trace!("got more packets");
+            trace!("got more packet batches in banking stage");
+            let (packets_received, packet_count_overflowed) = num_packets_received
+                .overflowing_add(packet_batch.iter().map(|batch| batch.packets.len()).sum());
```

Comment:
@pgarg66 hmm, how could `num_packets_received` overflow? it's usize, so effectively `u64` in our use case. and it's trusted in that sense that all of it is derived by `.sum()`-ing `.len()`s. So, i think oom could definitely happen before we hit overflowing condition here. maybe, could you write a test to demonstrate this edge case?

oh, i know i'm commenting on quite a bit old pr; it's just that I'm creating a pr touching this code. :)

Reply:
IMHO, it's fine to get rid of this overflowing handling; by definition, there can't be enough elements in the virtual address space to overflow a usize...

Reply:
It was mostly precautionary check. In theory it might happen (just looking at the math), but maybe never happen in practice.

Is this causing any bugs, or are you asking to remove it to simplify the code? In either case, it should be fine to remove it.

Reply:
FINALLY :)

fyi, I created a pr for this: #29715